pyuvsim/uvsim.py

```diff
@@ -538,7 +539,14 @@ def uvdata_to_task_iter(task_ids, input_uv, catalog, beam_list, beam_dict, Nsky_
         raise ValueError("If world keyword is set, it must be either 'moon' or 'earth'.")
     telescope = Telescope(input_uv.telescope_name, location, beam_list)
     freq_array = input_uv.freq_array * units.Hz
-    time_array = Time(input_uv.time_array, scale='utc', format='jd', location=telescope.location)
+    if hasmoon and isinstance(location, MoonLocation):
```

Comment:
this might be a little neater as:

```python
tclass = LTime if (hasmoon and isinstance(location, MoonLocation) else Time
time_array = tclass(....)
```

Reply:
Agreed, except that I will keep the if/else on separate lines because I find that much more readable.